### PR TITLE
[8.12.1] Upgrade to Lucene 9.9.2

### PR DESCRIPTION
**Pull request by ChrisHegarty**
Backport of Upgrade to Lucene 9.9.2 #104753

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/IndexVersions.java [hunk 1]
```diff
@@ -95,6 +95,8 @@ private static IndexVersion def(int id, Version luceneVersion) {
     public static final IndexVersion NORI_DUPLICATES = def(8_500_007, Version.LUCENE_9_9_0);
     public static final IndexVersion UPGRADE_LUCENE_9_9_1 = def(8_500_008, Version.LUCENE_9_9_1);
     public static final IndexVersion ES_VERSION_8_12_1 = def(8_500_009, Version.LUCENE_9_9_1);
+    public static final IndexVersion NEW_INDEXVERSION_FORMAT = def(8_501_00_0, Version.LUCENE_9_9_1);
+    public static final IndexVersion UPGRADE_LUCENE_9_9_2 = def(8_502_00_0, Version.LUCENE_9_9_2);
```
Comment (ChrisHegarty) [inline server/src/main/java/org/elasticsearch/index/IndexVersions.java:99]: @thecoop Now that we've got the _main_ changes sorted out, I wanna check how to proceed here with the 8.12.1 backport. I just aligned the versions. `NEW_INDEXVERSION_FORMAT` is not used at all, just backported it for alignment with _main_.

Comment (thecoop) [reply server/src/main/java/org/elasticsearch/index/IndexVersions.java:99]: 8.12.1 shouldn't know about any of the index versions beyond 8500010 for lucene 9.9.2 - and including them here will make 8.12 use those versions, which we don't want.

Comment (ChrisHegarty) [reply server/src/main/java/org/elasticsearch/index/IndexVersions.java]: removed the superfluous versions and added 8500010.   Should we include this, 8500010, in _main_ too for consistency ?

File: server/src/main/java/org/elasticsearch/index/IndexVersions.java [hunk 2]
```diff
@@ -95,6 +95,7 @@ private static IndexVersion def(int id, Version luceneVersion) {
     public static final IndexVersion NORI_DUPLICATES = def(8_500_007, Version.LUCENE_9_9_0);
     public static final IndexVersion UPGRADE_LUCENE_9_9_1 = def(8_500_008, Version.LUCENE_9_9_1);
     public static final IndexVersion ES_VERSION_8_12_1 = def(8_500_009, Version.LUCENE_9_9_1);
+    public static final IndexVersion UPGRADE_8_12_1_LUCENE_9_9_2 = def(8_500_010, Version.LUCENE_9_9_2);
```
Comment (ChrisHegarty) [inline server/src/main/java/org/elasticsearch/index/IndexVersions.java:98]: This name and value are the same as that in _main_.

**Review by thecoop — APPROVED** [review]
Looks good! This should be merged at the same time as #104753 though

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Pull request merged**